app/webpacker/components/RegistrationsV2/lib/RegistrationProvider.js

```diff
@@ -0,0 +1,97 @@
+import React, {
+  createContext, useCallback, useContext, useEffect, useMemo, useState,
+} from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { useDispatch } from '../../../lib/providers/StoreProvider';
+import { getSingleRegistration } from '../api/registration/get/get_registrations';
+import { setMessage } from '../Register/RegistrationMessage';
+import pollRegistrations from '../api/registration/get/poll_registrations';
+
+const REFETCH_INTERVAL = 3000;
+
+const RegistrationContext = createContext();
+
+export default function RegistrationProvider({ competitionInfo, userInfo, children }) {
+  const dispatch = useDispatch();
+  const [isProcessing, setIsProcessing] = useState(false);
```

Comment:
It would be helpful to distinguish between polling and processing (though the corresponding booleans are often equal). I think many uses of `isProcessing` (including this state) should be `isPolling`, and `startProcessing` should be `startPolling`. This hook/provider should still expose `isProcessing`, but via `pollingData && !pollingData.processing` rather than the (renamed) `isPolling` state.

I didn't really understand what the code was trying to do until I made this distinction in my head.